docs/sphinx/user_guide/feature/plugins.rst

```diff
@@ -0,0 +1,129 @@
+
+^^^^^^^^^^^
+Quick Start Guide
+^^^^^^^^^^^
+
+**Static**
+
+1. Build RAJA normally.
+
+2. Either use an ``#include`` statement within the code or compiler flags to load your plugin file with your project at compile time. A brief example of this would be something like ``g++ project.cpp plugin.cpp -lRAJA -fopenmp -ldl -o project``.
+
+3. When you run your project, your plugin should work!
+
+**Dynamic**
+
+1. Build RAJA normally.
+
+2. Compile your plugin to be shared object files with a .so extension. A brief example of this wouldbe something like ``g++ plugin.cpp -lRAJA -fopenmp -fPIC -shared -o plugin.so``.
+
+3. Set the environment variable ``RAJA_PLUGINS`` to be the path of your .so file. This can either be the path to its directory or to the shared object file itself. If the path is to a directory, it will attempt to load all .so files in that directory.
+
+4. When you run your project, your plugins should work!
+
+^^^^^^^^^^^
+Interfacing with Plugins
+^^^^^^^^^^^
+The RAJA Plugin API allows for limited interfacing between a project and a plugin. There are, however a couple functions that allow for this to take place. ``init_plugins`` and ``finalize_plugins``. Using one of these will call the corresponding ``init`` or ``finalize`` function inside of *every* currently loaded plugin. It's worth noting that plugins don't require either an init or finalize function by default.
+
+* ``RAJA::util::init_plugins();`` - Will call the ``init`` function of every currently loaded plugin.
+
+* ``RAJA::util::init_plugins("path/to/plugins");`` - Does the same as the above call to init_plugins, but will also dynamically load plugins located at the path specified.
+
+* ``RAJA::util::finalize_plugins();`` - Will call the ``finalize`` function of every currently loaded plugin. 
+
+
+------------
+Creating Plugins For RAJA
+------------
+
+Plugins take advantage of *polymorphism*, using ``RAJA::util::PluginStrategy`` as the parent and implementing the required functions for the API. An example implementation can be found at the bottom of this page.
+
+^^^^^^^^^^^
+Functions
+^^^^^^^^^^^
+The preLaunch and postLaunch functions are automatically called by RAJA before and after loop execution. This applies to RAJA's kernel and forall implementations.
+
+* ``void init(PluginOptions p) {}`` - runs on all plugins when the user makes a call to ``init_plugins``
+
+* ``void preCapture(PluginContext& p) {}`` - Will occur before capture of kernel/forall.
+
+* ``void postCapture(PluginContext& p) {}`` - Will occur after capture of kernel/forall.
+
+* ``void preLaunch(PluginContext& p) {}`` - Will occur before kernel/forall execution.
+
+* ``void postLaunch(PluginContext& p) {}`` - Will occur after kernel/forall execution.
+
+* ``void finalize() {}`` - runs on all plugins when the user makes a call to ``finalize_plugins``
```

Comment:
Function signatures.
Are these just signature or are they examples to the user? Should they have override or not?

Reply:
I'm not entirely sure if they need override or not, the idea there being that the user could just copy paste those in if they needed to. Whatever we end up settling on I'll update those to.

Reply:
I would think that they should indeed be overrided